### ISSUE TITLE
Symlinks mount error - setxattr operation not permitted

### ISSUE THREAD

**Issue by laped83** [containers/podman#14054]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Mounting a directory with symlinks can sometimes fail with the error: 
Error: setxattr /home/user/test/conf.d: operation not permitted

**Steps to reproduce the issue:**

1.
```
#!/bin/bash

TEST_DIR="${PWD}/test"
if [ ! -d "${TEST_DIR}" ]; then
  echo "Creating test dir structure"
  mkdir -p "${TEST_DIR}/etc"
  ln -sf /dev/null "${TEST_DIR}/conf.d"
  ln -sf /dev/null "${TEST_DIR}/etc/conf.d"
fi

while true; do
  if ! podman run --privileged -it --rm --volume "${TEST_DIR}":/test:Z fedora:36 pwd; then
    exit 0
  fi
done 

```
2. Wait for error

**Describe the results you received:**
Error: setxattr /home/user/test/conf.d: operation not permitted

**Describe the results you expected:**
output: /

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 4.0.3
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.3
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    variant: workstation
    version: "36"
  eventLogger: journald
  hostname: laped
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.3-302.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 1742487552
  memTotal: 16574898176
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.4-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.4
      commit: 6521fcc5806f20f6187eb933f9f45130c86da230
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 17179860992
  swapTotal: 17179860992
  uptime: 24h 46m 54.05s (Approximately 1.00 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/laa/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/laa/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 10
  runRoot: /run/user/1000/containers
  volumePath: /home/laa/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.3
  Built: 1648837314
  BuiltTime: Fri Apr  1 20:21:54 2022
  GitCommit: ""
  GoVersion: go1.18
  OsArch: linux/amd64
  Version: 4.0.3

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.0.3-1.fc36.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical fedora 36 install and a fedora 36 in VM/Boxes

**Comment by mheon** [comment]
@n1hility I recall this being mentioned before - is this fixed already in upstream?

**Comment by Luap99** [comment]
@rhatdan you fixed this, right?

**Comment by rhatdan** [comment]
Yes this should be fixed in podman 4.1
